setuptools_rust/build.py

```diff
@@ -407,45 +391,12 @@ def _py_limited_api(self) -> _PyLimitedApi:
 
     def _detect_rust_target(
         self, forced_target_triple: Optional[str] = None
-    ) -> Optional["_TargetInfo"]:
+    ) -> Optional[str]:
         assert self.plat_name is not None
-        cross_compile_info = _detect_unix_cross_compile_info()
```

Comment:
With an appropriate implementation of `setuptools_rust` that handles all build environment correctly, I still think there's value in a best-effort attempt to recognise the correct rust build target for use-cases like `crossenv`. My preference would be to not completely remove this, however the heavy-handedness with which it splats the build environment absolutely needs fixing.

Reply:
I've been thinking about the right implementation for `_detect_unix_cross_compile_info`, but don't have any good ideas right now. I'm also not terribly familiar with rust, but wound up with responsibility for `setuptools-rust` packaging in Void because it is required by our `python3-adblock` package as well as `python3-cryptography`.

For the reasons I've noted earlier, I think looking for `*_GNU_TYPE` variables in the target sysinfo is wrong because it will reflect the cross or native build environment of the Python interpreter rather than the compilation intent for the package being built with `setuptools-rust`. The best method here would be determining the native triple for the rust compiler and comparing that to `self.target`, but I don't know how to determine a native triple short of hacks like looking at `uname` output.

Reply:
> For the reasons I've noted earlier, I think looking for `*_GNU_TYPE` variables in the target sysinfo is wrong because it will reflect the cross or native build environment of the Python interpreter rather than the compilation intent for the package being built with `setuptools-rust`. The best method here would be determining the native triple for the rust compiler and comparing that to `self.target`, but I don't know how to determine a native triple short of hacks like looking at `uname` output.

That's what GNU autotools do though (look at `uname`). 

However, Yocto project does not generally trust any guessing that the components might try to do: we prefer to pass in all three triplets (host/build/target) explicitly via pre-written config files or command line switches.

Reply:
Specifically, here:
https://git.savannah.gnu.org/cgit/config.git/tree/config.guess

Reply:
Yeah, I think the better check for cross-compilation in this project is just "is `self.target` (read from `CARGO_BUILD_TARGET`) defined?"

Aside from checking the `*_GNU_TYPE` macros to pick cross info, all the `_detect_unix_cross_compile_info` does is attempt to determine the `cross_lib` path and set a default `PYO3_CROSS_LIB_DIR` based on it. I'd be OK with preserving that logic. (It determines the wrong value in our Void setup, but because I explicitly set the environment variable to the right value beforehand, it has no ill effect.)